### PR TITLE
Write SC for multiple output

### PR DESCRIPTION
**Pull request by andresaiello**
### Summary
- Smart contract for single input multiple output
- Refactor most used functions
- V2 for zevm swap contracts

### CODE REVIEW

File: packages/zevm-contracts/contracts/multi-output/ZetaMultiOutput.sol [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    constructor(address systemContractAddress) {
+        systemContract = SystemContract(systemContractAddress);
+    }
+
+    function registerDestinationToken(address destinationToken) external onlyOwner {
+        destinationTokens.push(destinationToken);
+        emit destinationRegistered(destinationToken);
+    }
+
+    function _transfersToDo(address zrc20) internal view returns (uint256) {
```
Comment (lucas-janon) [inline packages/zevm-contracts/contracts/multi-output/ZetaMultiOutput.sol:31]: I think `_pendingTransfers` works better for this.

File: packages/zevm-contracts/contracts/multi-output/ZetaMultiOutput.sol [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "../system/SystemContract.sol";
+import "../interfaces/IZRC20.sol";
+import "../interfaces/zContract.sol";
+import "../shared/BytesHelperLib.sol";
+import "../shared/SwapHelperLib.sol";
+
+interface ZetaMultiOutputErrors {
+    error NoTransfersToDo();
```
Comment (lucas-janon) [inline packages/zevm-contracts/contracts/multi-output/ZetaMultiOutput.sol:12]: And `NoPendingTransfers` for this one.

Comment (lucas-janon) [reply packages/zevm-contracts/contracts/multi-output/ZetaMultiOutput.sol:12]: Maybe `NoAvailableTransfers`?

File: packages/zevm-contracts/contracts/multi-output/ZetaMultiOutput.sol [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+}
+
+contract ZetaMultiOutput is zContract, Ownable, ZetaMultiOutputErrors {
+    SystemContract public immutable systemContract;
+    address[] public destinationTokens;
+
+    event destinationRegistered(address);
+    event withdrawal(address, uint256, address);
```
Comment (lucas-janon) [inline packages/zevm-contracts/contracts/multi-output/ZetaMultiOutput.sol:20]:
```suggestion
    event DestinationRegistered(address);
    event Withdrawal(address, uint256, address);
```

File: packages/zevm-contracts/scripts/zeta-swap/helpers.ts [hunk 4]
```diff
@@ -27,3 +28,21 @@ export const getBitcoinTxMemo = (zetaSwapAddress: string, destinationAddress: st
...
+    .hexlify(Number(chainId))
+    .slice(2)
+    .padStart(8, "0");
+  const rawMemo = `${destinationAddress.slice(2)}${paddedHexChainId}`;
+
+  const paddedMemo = rawMemo.padEnd(HashZero.length - 2, "0");
+  return `0x${paddedMemo}`;
+};
+
+export const getMultiOOutputForTest = (destinationAddress: string) => {
```
Comment (lucas-janon) [inline packages/zevm-contracts/scripts/zeta-swap/helpers.ts:43]: Is this one intended or a typo?

Comment (andresaiello) [reply packages/zevm-contracts/scripts/zeta-swap/helpers.ts]: typo. solved

**Review by lucas-janon — APPROVED** [review]
LGTM

**Pull request merged**